test/test_beltrami.py

```diff
@@ -0,0 +1,234 @@
+THE SOFTWARE.
+"""
+
+from dataclasses import dataclass
+import pytest
+
+import numpy as np
+
+from meshmode import _acf           # noqa: F401
+from arraycontext import pytest_generate_tests_for_array_contexts
+from meshmode.array_context import PytestPyOpenCLArrayContextFactory
+
+from pytential import bind, sym
+from pytential.symbolic.pde.beltrami import (
+        LaplaceBeltramiOperator, YukawaBeltramiOperator)
+
+import extra_int_eq_data as eid
+import logging
+logger = logging.getLogger(__name__)
+
+pytest_generate_tests = pytest_generate_tests_for_array_contexts([
+    PytestPyOpenCLArrayContextFactory,
+    ])
+
+
+# {{{ solutions
+
+def evaluate_spharm(actx, discr, m: int, n: int) -> np.ndarray:
+    assert discr.ambient_dim == 3
+
+    # {{{ get spherical coordinates
+
+    from arraycontext import thaw
+    x, y, z = thaw(discr.nodes(), actx)
+
+    theta = actx.np.arctan2(actx.np.sqrt(x**2 + y**2), z)
+    phi = actx.np.arctan2(y, x)
+
+    # }}}
+
+    # {{{ evaluate Y^m_n
+
+    from scipy.special import sph_harm      # pylint: disable=no-name-in-module
+    y_mn = []
+    for gtheta, gphi in zip(theta, phi):
+        result = sph_harm(m, n, actx.to_numpy(gphi), actx.to_numpy(gtheta))
+
+        y_mn.append(actx.from_numpy(result.real.copy()))
+
+    # }}}
+
+    return type(x)(actx, tuple(y_mn))
+
+
+@dataclass(frozen=True)
+class LaplaceBeltramiSolution:
+    name: str = "laplace"
+    radius: float = 1.0
+    m: int = 1
+    n: int = 3
+
+    @property
+    def eig(self):
+        # NOTE: eigenvalue of the -Laplacian on the sphere
+        return self.n * (self.n + 1) / self.radius ** 2
+
+    @property
+    def context(self):
+        return {}
+
+    def source(self, actx, discr):
+        return self.eig * evaluate_spharm(actx, discr, self.m, self.n)
+
+    def exact(self, actx, discr):
+        return evaluate_spharm(actx, discr, self.m, self.n)
+
+
+class YukawaBeltramiSolution(LaplaceBeltramiSolution):
+    name: str = "yukawa"
+    k: float = 1.0
+
+    @property
+    def context(self):
+        return {"k": self.k}
+
+    def source(self, actx, discr):
+        return (self.eig + self.k**2) * evaluate_spharm(actx, discr, self.m, self.n)
+
+# }}}
+
+
+# {{{ test_beltrami_convergence
+
+@pytest.mark.slowtest
+@pytest.mark.parametrize(("operator", "solution"), [
+    (LaplaceBeltramiOperator(3, precond="left"), LaplaceBeltramiSolution()),
+    (LaplaceBeltramiOperator(3, precond="right"), LaplaceBeltramiSolution()),
+    (YukawaBeltramiOperator(3, precond="left"), YukawaBeltramiSolution()),
+    ])
+def test_beltrami_convergence(actx_factory, operator, solution, visualize=False):
+    if visualize:
+        logging.basicConfig(level=logging.INFO)
+    actx = actx_factory()
+
+    radius = 1
+    case = eid.SphereTestCase(
+            target_order=5,
+            qbx_order=5,
+            source_ovsmp=8,
+            fmm_order=False, fmm_tol=None, fmm_backend=None,
```

Comment:
Tried turning on FMM here, but there's a lot of operators and composition of operators, so it cranks for a long time in code generation. Any preference? The test is marked as `slowtest`, so it won't have an effect on the CI.

Reply:
Would any setting allow us to run this as part of CI? How long does it take?